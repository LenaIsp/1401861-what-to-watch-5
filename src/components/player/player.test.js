
import React from 'react';
import renderer from 'react-test-renderer';
import Player from './player';
import {TEST_MOCKS} from "../../test-mock.js";

it(`Player is rendered correctly`, () => {

  const tree = renderer.create(
      <Player
        films={TEST_MOCKS.movies}
        onReplayButtonClick = {TEST_MOCKS.noop}
        routes = {TEST_MOCKS.routes}
        handleClickPlay = {TEST_MOCKS.noop}
        handleClickPause = {TEST_MOCKS.noop}
        handleClickFullScreen = {TEST_MOCKS.noop}
        player = {`play`}
        duration = {50}
        currentTime = {50}
      >
        <video />
      </Player>, {
        createNodeMock: () => {
          return {};
        }
      }).toJSON();

  expect(tree).toMatchSnapshot();
});
