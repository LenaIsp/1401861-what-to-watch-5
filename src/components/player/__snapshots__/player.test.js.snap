// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player is rendered correctly 1`] = `
<div
  className="player"
>
  <video />
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
          value={100}
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "100%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        00:00:00
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          fill="none"
          height="19"
          viewBox="0 0 19 19"
          width="19"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M0 0L19 9.5L0 19V0Z"
            fill="#EEE5B5"
            fillRule="evenodd"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <div
        className="player__name"
      >
        Bronson
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          fill="none"
          height="27"
          viewBox="0 0 27 27"
          width="27"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M23.8571 0H16V3.14286H23.8571V11H27V3.14286V0H23.8571Z"
            fill="#FFF9D9"
            fillOpacity="0.7"
            fillRule="evenodd"
          />
          <path
            clipRule="evenodd"
            d="M27 23.8571V16H23.8571V23.8571H16V27H23.8571H27L27 23.8571Z"
            fill="#FFF9D9"
            fillOpacity="0.7"
            fillRule="evenodd"
          />
          <path
            clipRule="evenodd"
            d="M0 3.14286L0 11H3.14286L3.14286 3.14286L11 3.14286V0H3.14286H0L0 3.14286Z"
            fill="#FFF9D9"
            fillOpacity="0.7"
            fillRule="evenodd"
          />
          <path
            clipRule="evenodd"
            d="M3.14286 27H11V23.8571H3.14286L3.14286 16H0L0 23.8571V27H3.14286Z"
            fill="#FFF9D9"
            fillOpacity="0.7"
            fillRule="evenodd"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;
